lib/csetgrp.gi

```diff
@@ -988,9 +1052,13 @@ local c, flip, maxidx, refineChainActionLimit, cano, tryfct, p, r, t,
 	fi;
       end;
 
-      for i in TryMaximalSubgroupClassReps(G:cheap) do
-	if Index(G,i)<maxidx(c) and Index(G,i)<=avoidlimit then
+      r:=Filtered(TryMaximalSubgroupClassReps(G:cheap),
+        x->Index(G,x)<=5*avoidlimit);
+      SortBy(r,a->-Size(a));
+      for i in r do
+	if Index(G,i)<maxidx(c) then
 	  p:=Intersection(a,i);
+          AddSet(doneidx,Index(a,p));
```

Comment:
indentation

Reply:
I've globally replaced tabs and now its OK.